sql/core/src/main/java/org/apache/spark/sql/execution/datasources/parquet/SpecificParquetRecordReaderBase.java

```diff
@@ -93,48 +93,17 @@ public void initialize(InputSplit inputSplit, TaskAttemptContext taskAttemptCont
       throws IOException, InterruptedException {
     Configuration configuration = taskAttemptContext.getConfiguration();
     ParquetInputSplit split = (ParquetInputSplit)inputSplit;
+    Preconditions.checkState(split.getRowGroupOffsets() == null,
+            "rowGroupOffsets in ParquetInputSplit should be null.");
     this.file = split.getPath();
-    long[] rowGroupOffsets = split.getRowGroupOffsets();
-
-    ParquetMetadata footer;
-    List<BlockMetaData> blocks;
-
-    // if task.side.metadata is set, rowGroupOffsets is null
-    if (rowGroupOffsets == null) {
```

Comment:
After rethinking @LuciferYang, actually this is aimed to be ported to Apache Parquet, see the comments above:

```
 * TODO: move this to the parquet-mr project. There are performance benefits of doing it
 * this way, albeit at a higher cost to implement. This base class is reusable.
```

It's being tracked at [PARQUET-131](https://issues.apache.org/jira/browse/PARQUET-131). Shall we just simply add a comment that this code is not used in Apache Spark for now?

Reply:
Add a comment and revert the code change?

Reply:
But PARQUET-131 seems to have made no progress in the last three years ... ,  anyone really do it ?

I'm just worried about whether this TODO will really be completed :)

Reply:
Yeah, just simply adding a comment because this is planned to move to Parquet. It's stuck but I think we should port it in long run anyway.

Reply:
OK ~

Reply:
done ~